onetl/file/format/avro.py

```diff
@@ -189,6 +192,128 @@ def apply_to_writer(self, writer: DataFrameWriter) -> DataFrameWriter:
             options["avroSchema"] = json.dumps(self.schema_dict)
         return writer.format(self.name).options(**options)
 
+    def parse_column(self, column: str | Column) -> Column:
+        """
+        Parses an Avro binary column into a structured Spark SQL column using Spark's
+        `from_avro <https://spark.apache.org/docs/latest/api/python/reference/pyspark.sql/api/pyspark.sql.avro.functions.from_avro.html>`_ function, based on the schema provided within the class.
+
+        .. note::
+
+            Can be used only with Spark 3.x+
+
+        Parameters
+        ----------
+        column : str | Column
+            The name of the column or the Column object containing Avro binary data to parse.
+
+        Returns
+        -------
+        Column
+            A new Column object with data parsed from Avro binary to the specified structured format.
+
+        Examples
+        --------
+        .. code:: python
+
+            from pyspark.sql import SparkSession
+
+            from onetl.file.format import Avro
+
+            spark = SparkSession.builder.appName("AvroParsingExample").getOrCreate()
+            schema_dict = {
+                "type": "record",
+                "name": "Person",
+                "fields": [{"name": "name", "type": "string"}],
+            }
+            avro = Avro(schema_dict=schema_dict)
+            df = spark.createDataFrame([("bytes_data_here",)], ["avro_data"])
+
+            parsed_df = df.select(avro.parse_column("avro_data"))
+            parsed_df.show()
+
+        """
+        from pyspark.sql import Column, SparkSession  # noqa: WPS442
+        from pyspark.sql.avro.functions import from_avro
+        from pyspark.sql.functions import col
+
+        self.check_if_supported(SparkSession.getActiveSession())
+
+        if isinstance(column, Column):
+            column_name = column._jc.toString()  # noqa: WPS437
+        else:
+            column_name, column = column, col(column).cast("binary")
+
+        if self.schema_dict:
+            schema_json = json.dumps(self.schema_dict)
+        elif self.schema_url:
+            response = requests.get(self.schema_url)  # noqa: S113
+            schema_json = response.text
+        else:
+            raise ValueError("No schema defined in Avro class instance.")
```

Comment:
Думаю стоит поменять текст сообщения на примерно такой:
```suggestion
            raise ValueError("Avro.parse_column can be used only with defined `schema_dict` or `schema_url`")
```